Clockio/app/src/main/java/com/natasha/clockio/presence/ui/fragment/ImageFragment.kt

```diff
@@ -1,65 +1,202 @@
 import android.net.Uri
-import androidx.lifecycle.ViewModelProviders
 import android.os.Bundle
 import android.util.Log
 import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.Toast
+import androidx.lifecycle.ViewModelProvider
+import cn.pedant.SweetAlert.SweetAlertDialog
 import com.bumptech.glide.Glide
+import com.google.gson.Gson
 
 import com.natasha.clockio.R
-import kotlinx.android.synthetic.main.image_fragment.*
+import com.natasha.clockio.base.model.BaseResponse
+import com.natasha.clockio.base.model.DataResponse
+import com.natasha.clockio.location.LocationModel
+import com.natasha.clockio.location.LocationViewModel
+import com.natasha.clockio.presence.service.request.CheckinRequest
+import com.natasha.clockio.presence.viewModel.ImageViewModel
+import com.natasha.clockio.presence.viewModel.PresenceViewModel
+import dagger.android.support.AndroidSupportInjection
+import kotlinx.android.synthetic.main.fragment_image.*
+import okhttp3.ResponseBody
 import java.io.File
+import java.util.*
+import javax.inject.Inject
 
 class ImageFragment : Fragment() {
 
-    companion object {
-        fun newInstance() = ImageFragment()
-        private val TAG = ImageFragment::class.java.simpleName
-    }
+  companion object {
+    fun newInstance() = ImageFragment()
+    private val TAG = ImageFragment::class.java.simpleName
+  }
+
+  @Inject lateinit var sharedPref: SharedPreferences
+//  @Inject lateinit var gson: Gson
+  @Inject lateinit var factory: ViewModelProvider.Factory
+  private lateinit var imageViewModel: ImageViewModel
+  private lateinit var locationViewModel: LocationViewModel
+  private lateinit var presenceViewModel: PresenceViewModel
+  private lateinit var imagePath: String
+  private var employeeId: String? = null
+  var location: LocationModel = LocationModel(0.0, 0.0)
+
+  override fun onAttach(context: Context) {
+    AndroidSupportInjection.inject(this)
+    super.onAttach(context)
+  }
+
+  override fun onCreateView(
+      inflater: LayoutInflater, container: ViewGroup?,
+      savedInstanceState: Bundle?
+  ): View? {
+    return inflater.inflate(R.layout.fragment_image, container, false)
+  }
+
+  override fun onActivityCreated(savedInstanceState: Bundle?) {
+    super.onActivityCreated(savedInstanceState)
+    imageViewModel = ViewModelProvider(this, factory).get(ImageViewModel::class.java)
+    locationViewModel = ViewModelProvider(this, factory).get(LocationViewModel::class.java)
+    presenceViewModel = ViewModelProvider(this, factory).get(PresenceViewModel::class.java)
+    observeProgress()
+    observeEvents()
+    observeCheckInResult()
+  }
 
-    private lateinit var viewModel: ImageViewModel
-    private lateinit var imagePath: String
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    employeeId = sharedPref.getString("id", null)
```

Comment:
put this String in Constants, so other files can access these also

Reply:
done